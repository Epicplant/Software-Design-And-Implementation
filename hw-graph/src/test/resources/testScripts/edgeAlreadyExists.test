# Tests that adding an edge that already exists doesn't change anything

# Creates a new graph
CreateGraph newGraph

# Adds a new node n1
AddNode newGraph n1

# Adds a new node n2
AddNode newGraph n2

# Adds an edge between n1 and n2 of weight 0
AddEdge newGraph n1 n2 0

# Adds a duplicate edge between n1 and n2 of weight 0
AddEdge newGraph n1 n2 0

# Lists children of n1 which should only have one copy of the edge
ListChildren newGraph n1