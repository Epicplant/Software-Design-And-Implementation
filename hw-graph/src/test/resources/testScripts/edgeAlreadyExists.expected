# Tests that adding an edge that already exists doesn't change anything

# Creates a new graph
created graph newGraph

# Adds a new node n1
added node n1 to newGraph

# Adds a new node n2
added node n2 to newGraph

# Adds an edge between n1 and n2 of weight 0
added edge 0 from n1 to n2 in newGraph

# Adds a duplicate edge between n1 and n2 of weight 0
added edge 0 from n1 to n2 in newGraph

# Lists children of n1 which should only have one copy of the edge
the children of n1 in newGraph are: n2(0)